src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,127 @@
+package duke.storage;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.util.ArrayList;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.TaskList;
+import duke.task.ToDo;
+
+/**
+ * Storage class that save the data
+ */
+public class Storage {
+    private final String filePath;
+
+    /**
+     * Instantiates the Storage class
+     *
+     * @param directoryName the name of the directory
+     * @param fileName the name of the file
+     */
+    public Storage(String directoryName, String fileName) {
+        this.filePath = directoryName + "/" + fileName;
+        // Make new file to store
+        try {
+            File directory = new File(directoryName);
+
+            // Make the directory if not exist
+            if (!directory.exists()) {
+                directory.mkdir();
+            }
+
+            File file = new File(this.filePath);
+
+            // Make the file if not exist
+            if (file.exists()) {
+                loadTasks();
+            } else {
+                file.createNewFile();
+            }
+        } catch (IOException e) {
+            System.out.println("Failed to load tasks: " + e.getMessage());
+        }
+    }
+
+    /**
+     * Loads the tasks from the storage
+     *
+     * @return the tasks loaded
+     * @throws IOException when the task stored is invalid (usually the date)
+     */
+    public ArrayList<Task> loadTasks() throws IOException {
+        ArrayList<Task> loadedTasks = new ArrayList<>();
+        BufferedReader reader = new BufferedReader(new FileReader(this.filePath));
+        String line = reader.readLine();
+
+        // Adds each task
+        while (line != null) {
+            Task currentTask = readTask(line);
+            loadedTasks.add(currentTask);
+            line = reader.readLine();
+        }
+
+        reader.close();
+        return loadedTasks;
+    }
+
+    /**
+     * Reads a task stored in the storage
+     *
+     * @param taskStoredInString the string representative of the task stored in the storage
+     * @return the task
+     * @throws IOException if the task read is invalid
+     */
+    public static Task readTask(String taskStoredInString) throws IOException {
+        String[] taskInStringSplit = taskStoredInString.split(" \\| ");
```

Comment:
Perhaps the name of variable `taskInStringSplit` could be improved by reordering to `splitTaskInString`?